node/node.go

```diff
@@ -249,6 +250,17 @@ func (n *Node) CreateLedgerChannel(Counterparty types.Address, ChallengeDuration
 		// Appdata implicitly zero
 	)
 
+	// Check store to see if there is an existing channel with this counterparty
+	channelExists, err := directfund.ChannelsExistWithCounterparty(Counterparty, n.store.GetChannelsByParticipant, n.store.GetConsensusChannel)
```

Comment:
This is a synchronous request validation check that can immediately return an error to the client. Otherwise, no error is returned to the client and asynchronous request processing begins